packages/@ember/-internals/glimmer/lib/helpers/unique-id.ts

```diff
@@ -29,7 +29,7 @@ import { createConstRef, Reference } from '@glimmer/reference';
 import { internalHelper } from './internal-helper';
 
 export default internalHelper((): Reference<string> => {
-  return createConstRef(uniqueId(), 'unique-id');
+  return createConstRef(`em-${uniqueId()}`, 'unique-id');
```

Comment:
I think the comment below is suggesting that we _specifically_ do not want to do this. In the past, things like `id="ember-*"` became de-facto relied upon and impossible to change. The proposed change has a similar issue in that people could come to rely on the id having a particular format (starting with `em-`).

The original authors of the code chose to use a UUID v4 here to avoid this problem. Arguably, the UUID v4 format itself – `xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx` – could become relied upon, though it's much harder than a fixed prefix. Perhaps we should have made it all random characters, but perhaps there were more considerations that went into it and in any case that's probably out of scope for what you are trying to do/fix here.

Instead, I figured we could have preserved the UUID v4-ness _and_ still resolve the issue you raised here, with this One Magical Tweak™:

```diff
 function uniqueId() {
   // @ts-expect-error this one-liner abuses weird JavaScript semantics that
   // TypeScript (legitimately) doesn't like, but they're nonetheless valid and
   // specced.
-   return ([1e7] + -1e3 + -4e3 + -8e3 + -1e11).replace(/[018]/g, (a) =>
-    (a ^ ((Math.random() * 16) >> (a / 4))).toString(16)
+   return ([3e7] + -1e3 + -4e3 + -2e3 + -1e11).replace(/[0-3]/g, (a) =>
+    ((a * 4) ^ ((Math.random() * 16) >> (a & 2))).toString(16)
   );
 }
```

Essentially, this generates a valid UUID v4, but fixes the first character to one of `c/d/e/f`. Since it always starts with a "letter" it should accomplish the same thing while not introducing a fixed prefix, respecting the original authors' wishes.

If anyone is interested I can unpack how it works 😬, but it requires first unpacking what the [original version](https://gist.github.com/jed/982883) was trying to do and tweaking it slightly. But it basically does the same thing, just also restricting the first character.

We should probably add a test for this (asserting that the result can be a valid selector). It's a bit challenging without exposing the guts of this, but I figured calling the helper/running the assertions say 1000 times in a test is probably good enough proof that it's working?